mods/cnc/maps/gdi05a/gdi05a.lua

```diff
@@ -0,0 +1,230 @@
+	gdi.MarkCompletedObjective(gdiObjective1)
+	
+	Attack()
+end
+
+SetupWorld = function()
+	Utils.Do(ActorRemovals[Map.Difficulty], function(unit)
+		unit.Destroy()
+	end)
+
+	Reinforcements.Reinforce(gdi, GdiTanks, { GdiTankEntry.Location, GdiTankRallyPoint.Location }, DateTime.Seconds(1), function(actor) actor.Stance = "Defend" end)
+	Reinforcements.Reinforce(gdi, GdiApc, { GdiApcEntry.Location, GdiApcRallyPoint.Location }, DateTime.Seconds(1), function(actor) actor.Stance = "Defend" end)
+	Reinforcements.Reinforce(gdi, GdiInfantry, { GdiInfantryEntry.Location, GdiInfantryRallyPoint.Location }, 15, function(actor) actor.Stance = "Defend" end)
+
+	Trigger.OnEnteredFootprint(GdiBaseDiscoveryTrigger, function(actor, id)
+		if actor.Owner == gdi then
+			DiscoverGdiBase()
+			Trigger.RemoveFootprintTrigger(id)
+		end
+	end)
+
+	Utils.Do(Map.NamedActors, function(actor)
+		if actor.Owner == nod and actor.HasProperty("StartBuildingRepairs") then
+			Trigger.OnDamaged(actor, function(building)
+				if building.Owner == nod and building.Health < RepairThreshold[Map.Difficulty] * building.MaxHealth then
+					building.StartBuildingRepairs()
+				end
+			end)
+		end
+	end)
+
+	Trigger.OnAllKilled(NodSams, function()
+		gdi.MarkCompletedObjective(gdiObjective2)
+		Actor.Create("airstrike.proxy", true, { Owner = gdi })
+	end)
+
+	GdiHarv.Stop()
+	NodHarv.FindResources()
+	if Map.Difficulty ~= "Easy" then
+		Trigger.OnDamaged(NodHarv, function()
+			Utils.Do(nod.GetGroundAttackers(), function(unit)
+				unit.AttackMove(NodHarv.Location)
+				if Map.Difficulty == "Hard" then
+					unit.Hunt()
+				end
+			end)
+		end)
+	end
+
+	Trigger.AfterDelay(DateTime.Seconds(45), Grd1Action)
+	Trigger.AfterDelay(DateTime.Minutes(3), Grd2Action)
+	Grd3Action()
+end
+
+WorldLoaded = function()
+	gdi = Player.GetPlayer("GDI")
+	nod = Player.GetPlayer("Nod")
+
+	Trigger.OnObjectiveAdded(gdi, function(p, id)
+		Media.DisplayMessage(p.GetObjectiveDescription(id), "New " .. string.lower(p.GetObjectiveType(id)) .. " objective")
+	end)
+	Trigger.OnObjectiveCompleted(gdi, function(p, id)
+		Media.DisplayMessage(p.GetObjectiveDescription(id), "Objective completed")
+	end)
+	Trigger.OnObjectiveFailed(gdi, function(p, id)
+		Media.DisplayMessage(p.GetObjectiveDescription(id), "Objective failed")
+	end)
+
+	Trigger.OnPlayerLost(gdi, function()
+		Media.PlaySpeechNotification(player, "Lose")
+	end)
+
+	Trigger.OnPlayerWon(gdi, function()
+		Media.PlaySpeechNotification(player, "Win")
+	end)
+
+	nodObjective = nod.AddPrimaryObjective("Destroy all GDI troops")
+	gdiObjective1 = gdi.AddPrimaryObjective("Find the GDI base")
+	gdiObjective2 = gdi.AddSecondaryObjective("Destroy all SAM sites to receive air support")
+
+	SetupWorld()
+
+	Camera.Position = GdiTankRallyPoint.CenterPosition
+
+	Media.PlayMusic()
+end
+
+Tick = function()
+	if gdi.HasNoRequiredUnits() then
+		if DateTime.GameTime > 2 then
+			nod.MarkCompletedObjective(nodObjective)
```

Comment:
[no, it won't](https://github.com/OpenRA/OpenRA/blob/bleed/OpenRA.Mods.Common/Traits/Player/MissionObjectives.cs#L230)